### ISSUE TITLE
When already in a map view, clicking "Publish" pompted me to save my map

### ISSUE THREAD

**Issue by chriswhong** [CartoDB/cartodb#5164]
This one will be hard to repro, but basically when I was trying to show the direct and embed links for a multi-layer map, I clicked "Publish" and the screen that normally shows up when adding a 2nd layer to a dataset's map view.  

@jatorre was in the audience and may be able to explain better.

**Comment by viddo** [comment]
From what you describe [this check that the vis model represents a map](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/old_common/header.js#L144-L157) must have returned false and thus entered the else-clause to "create the vis (map) first". 

I have my primary suspects, when I migrated the old code I continued using the [existing way of handling this](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/common/dialogs/create_vis_first/create_vis_first_view.js#L61), which does a bunch of mutations locally (updating model(s), change browser URL etc.). 

IMHO we would better of removing this complexity and simply redirect the user to the new created vis URL (just like when you open a vis from the dashboard) to avoid this kind of issues in the future, and remove some code in the process. @javisantana what is your opinion on this?

Btw, thx for reporting, and I understand that this bug might not be the easiest to reproduce, but if it can't be reproduced consistently (i.e. have a series of steps-to-reproduce) at least try to give as much info as possible (dataset(s) used, what modifications were done up to the point etc.), otherwise it will be very hard for us to find and know that we actually fixed the issue.
